src/proto/streams/recv.rs

```diff
@@ -590,7 +590,7 @@ impl Recv {
     /// Handle remote sending an explicit RST_STREAM.
     pub fn recv_reset(&mut self, frame: frame::Reset, stream: &mut Stream) {
         // Notify the stream
-        stream.state.recv_reset(frame.reason());
+        stream.state.recv_reset(frame.reason(), stream.is_pending_send);
```

Comment:
I'd love to eventually make the `state` private. It reads weird that we need to pass in extra information about the stream itself. (Not your fault, I'm just pining for a better future some day :D).

Reply:
As far as I can tell, there doesn't *appear* to be any harm in making the forcing the transition for *all* `Closed(Cause::EndStream)` states on receipt of RST_STREAM regardless of whether or not the stream has pending send; but @carllerche suggested that we only transition to `Closed(Cause::Proto)` if the stream is queued.

Reply:
I prefer to stay on the side of conservative and only make changes that we know are needed.

If there are no queued send frames, then the stream state is actually closed and the received reset should be ignored (I think this is already done)

Reply:
I don't think I was clear. I get why we're using the queued information, that's fine! We don't want to trigger a reset error for a stream the user has already closed locally.

I was more concerned that it's weird that we pass information of the stream as an argument onto the state property. I'd want to one day change the methods on the `state` to be methods on `stream` instead.

In other words, ignore me. I'm just sleep deprived. Move along, nothing to see here!